src/record.rs

```diff
@@ -80,6 +80,52 @@ impl Meta {
     }
 }
 
+pub(crate) struct PartiallySerializedRecord {
+    buf: Vec<u8>,
+    offset_pos: usize,
+    header_checksum_pos: usize,
+    header_start: usize,
+    header_len: usize,
+}
+
+impl PartiallySerializedRecord {
+    fn new(
+        buf: Vec<u8>,
+        offset_pos: usize,
+        header_checksum_pos: usize,
+        header_start: usize,
+        header_len: usize,
+    ) -> Self {
+        Self {
+            buf,
+            offset_pos,
+            header_checksum_pos,
+            header_start,
+            header_len,
+        }
+    }
+
+    pub(crate) fn into_serialized_with_header_checksum(
+        self,
+        blob_offset: u64,
+    ) -> Result<(Vec<u8>, u32)> {
+        let Self {
+            mut buf,
+            offset_pos,
+            header_checksum_pos,
+            header_start,
+            header_len,
+        } = self;
+        let offset_slice = &mut buf[offset_pos..];
+        bincode::serialize_into(offset_slice, &blob_offset)?;
```

Comment:
I suggest to use `offset_slice.copy_from_slice(&blob_offset.to_le_bytes());`
```suggestion
        offset_slice.copy_from_slice(&blob_offset.to_le_bytes());
```

Reply:
It panics at `source slice length does not match destination slice length`, which is not checked in serialization function (it just writes as much as possible). Do you think we should take len and obtain exact slice?

Reply:
Yes, the size of the type can be calculated with `std::mem::size_of::<type>()`:
```
offset_slice[..std::mem::size_of::<u64>()].copy_from_slice(&blob_offset.to_le_bytes());
```